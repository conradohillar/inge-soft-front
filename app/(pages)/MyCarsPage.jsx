import { SafeAreaView } from "react-native-safe-area-context";
import Header from "../../components/Header";
import { FlatList, View, Text, ScrollView, Image } from "react-native";
import CarCard from "../../components/CarCard";
import { Button, XStack } from "tamagui";
import { Link, useLocalSearchParams, useRouter } from "expo-router";
import icons from "../../constants/icons"
import { QueryClient, QueryClientProvider, useQuery, useMutation } from '@tanstack/react-query'
import {LOCAL_IP} from '@env'
import LoadingPage from '../(pages)/LoadingPage'
import ErrorPage from "./ErrorPage";
import * as SecureStore from 'expo-secure-store';
import React, { useState, useEffect } from "react";
import { useFocusEffect } from '@react-navigation/native';

const queryClient = new QueryClient()

export default function MyCarsPage() {

    return (
        <QueryClientProvider client={queryClient} >
            <Content />
        </QueryClientProvider>
    )
}

function Content() {
    const [token, setToken] = useState(null);


    useEffect(() => {
        const fetchToken = async () => {
            try {
                const storedToken = await SecureStore.getItemAsync('token');
                if (storedToken) {
                    setToken(storedToken);
                }
            } catch (error) {
                console.error('Error fetching token from SecureStore', error);
            }
        };

        fetchToken();
    }, []);


    const url = `http://${LOCAL_IP}:8000/users/mycars`
    
    const headers = {
        Authorization: `Bearer ${token}`,
        'Content-Type': 'application/json'
    };

    

    const { isPending, error, data } = useQuery({
        queryKey: ['fetchMycars'],
        queryFn: () =>
            fetch(url, {headers} ).then((res) =>
                res.json(),
            ),
            enabled: !!token,

    });
    
    
    if (isPending) {
        return <LoadingPage />
    }

    if (error) {
        return <ErrorPage />
    }


    
    const renderItem = ({ item }) => {
        return (
            <CarCard model={item.model} plate={item.plate} />
        );

    }
    return (
        <SafeAreaView className="w-full h-full bg-background">
            <Header />
            <View className="items-start mt-5 ml-4">
                <Link href="/(tabs)/profile" asChild>
                    <Button className="h-9 w-9 bg-background rounded-xl">
                        <Image source={icons.arrowleft} className="h-7 w-7" resizeMode="contain" />
                    </Button>
                </Link>
            </View>
            <XStack className="items-center justify-center mb-7">
                <Text className="text-4xl font-qbold text-black">MIS AUTOS</Text>
            </XStack>
            { data!= undefined && <FlatList
                className="w-full"
                data={data}
                keyExtractor={item => item.plate}
                renderItem={renderItem}
            />}
            <XStack className="items-center justify-center my-10">
                <Text className="text-lg text-gray-600 font-qsemibold">Agreg√° un auto nuevo</Text>
                <Link href="/(pages)/AddCarPage" asChild>
                    <Button className="h-8 w-8 bg-background rounded-xl ml-3 mt-1">
                        <Image source={icons.add} className="h-6 w-6" resizeMode="contain" />
                    </Button>
                </Link>
            </XStack>
        </SafeAreaView>
    );

}