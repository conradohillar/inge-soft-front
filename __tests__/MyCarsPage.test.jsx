import React from 'react';
import { render, waitFor, fireEvent } from '@testing-library/react-native';
import MyCarsPage from '../app/(pages)/MyCarsPage';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { TamaguiProvider } from "tamagui";
import config from "../tamagui.config";
import axios from 'axios';

jest.mock('axios');


const queryClient = new QueryClient();
const Provider = ({ children }) => (
    <TamaguiProvider config={config}>
        <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
    </TamaguiProvider>
);

describe('MyCarsPage', () => {


    it('renders loading state initially', () => {
        const { getByText } = render(<Provider><MyCarsPage /></Provider>);

        expect(getByText('Estamos cargando')).toBeTruthy();

    });

    const cars = [
        { model: 'Toyota', plate: 'ABC123' },
        { model: 'Honda', plate: 'XYZ789' },
    ];


    axios.get.mockResolvedValueOnce({ data: cars });

    it('renders list of cars', async () => {


        const { getByText } = render(<Provider><MyCarsPage /></Provider>);


        await waitFor(() => {
            expect(getByText('Toyota')).toBeTruthy();
            expect(getByText('Honda')).toBeTruthy();
        });
    });
    describe('my cars page Snapshot', () => {
        it('deberÃ­a coincidir con la snapshot', () => {
            const tree = render(<Provider><MyCarsPage /></Provider>).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

});
