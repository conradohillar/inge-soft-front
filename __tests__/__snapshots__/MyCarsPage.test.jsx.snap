// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyCarsPage my cars page Snapshot deber√≠a coincidir con la snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    [
      {
        "width": "100%",
      },
      {
        "height": "100%",
      },
      {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#ccc",
        "borderBottomWidth": 2,
        "borderStyle": "solid",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "minHeight": "8%",
        "paddingLeft": 16,
        "paddingRight": 16,
        "width": "100%",
      }
    }
  >
    <View
      focusVisibleStyle={{}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="button"
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomColor": "transparent",
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderBottomWidth": 1,
          "borderLeftColor": "transparent",
          "borderLeftWidth": 1,
          "borderRightColor": "transparent",
          "borderRightWidth": 1,
          "borderStyle": "solid",
          "borderTopColor": "transparent",
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
          "borderTopWidth": 1,
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "height": 40,
          "justifyContent": "center",
          "paddingLeft": 18,
          "paddingRight": 18,
          "width": 40,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          [
            {
              "height": 32,
            },
            {
              "width": 32,
            },
          ]
        }
      />
    </View>
    <Text
      href="/home"
      onPress={[Function]}
      role="link"
      style={
        [
          {
            "fontSize": 48,
            "lineHeight": 48,
          },
          {
            "fontFamily": "Quicksand-Bold",
          },
          {
            "color": "#59A58A",
          },
          {
            "marginTop": 8,
          },
        ]
      }
    >
      rydio
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "flex-start",
        },
        {
          "marginTop": 20,
        },
        {
          "marginLeft": 16,
        },
      ]
    }
  >
    <View
      focusVisibleStyle={{}}
      focusable={true}
      href="/profile"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="link"
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomColor": "transparent",
          "borderBottomLeftRadius": 12,
          "borderBottomRightRadius": 12,
          "borderBottomWidth": 1,
          "borderLeftColor": "transparent",
          "borderLeftWidth": 1,
          "borderRightColor": "transparent",
          "borderRightWidth": 1,
          "borderStyle": "solid",
          "borderTopColor": "transparent",
          "borderTopLeftRadius": 12,
          "borderTopRightRadius": 12,
          "borderTopWidth": 1,
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "height": 36,
          "justifyContent": "center",
          "paddingLeft": 18,
          "paddingRight": 18,
          "width": 36,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          [
            {
              "height": 28,
            },
            {
              "width": 28,
            },
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 28,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 36,
            "lineHeight": 40,
          },
          {
            "fontFamily": "Quicksand-Bold",
          },
          {
            "color": "#000",
          },
        ]
      }
    >
      MIS AUTOS
    </Text>
  </View>
  <RCTScrollView
    data={
      [
        {
          "model": "Toyota",
          "plate": "ABC123",
        },
        {
          "model": "Honda",
          "plate": "XYZ789",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    style={
      [
        {
          "width": "100%",
        },
      ]
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            [
              {
                "width": "100%",
              },
              {
                "alignItems": "center",
              },
              {
                "marginBottom": 12,
                "marginTop": 12,
              },
            ]
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#eee",
                "borderColor": "#aaa",
                "borderRadius": 40,
                "borderWidth": 2,
                "height": 150,
                "paddingHorizontal": 30,
                "paddingTop": 20,
                "width": "85%",
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "column",
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "flex-start",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  style={
                    {
                      "flexDirection": "column",
                    }
                  }
                >
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#000",
                          },
                          {
                            "fontSize": 24,
                            "lineHeight": 32,
                          },
                          {
                            "fontFamily": "Quicksand-Bold",
                          },
                        ]
                      }
                    >
                      Toyota
                    </Text>
                    <Image
                      resizeMode="contain"
                      source={1}
                      style={
                        [
                          {
                            "height": 40,
                          },
                          {
                            "width": 40,
                          },
                          [
                            {
                              "marginLeft": 20,
                            },
                          ],
                        ]
                      }
                      tintColor="#bbb"
                    />
                  </View>
                  <Text
                    style={
                      [
                        {
                          "fontSize": 16,
                          "lineHeight": 24,
                        },
                        {
                          "fontFamily": "Quicksand-SemiBold",
                        },
                        [
                          {
                            "marginTop": 12,
                          },
                        ],
                      ]
                    }
                  >
                    Patente: 
                    ABC123
                  </Text>
                </View>
              </View>
              <View
                style={
                  [
                    {
                      "alignItems": "flex-end",
                    },
                    {
                      "marginBottom": 12,
                    },
                  ]
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "backgroundColor": "#eee",
                    }
                  }
                >
                  <Image
                    resizeMode="contain"
                    source={1}
                    style={
                      [
                        {
                          "height": 28,
                        },
                        {
                          "width": 28,
                        },
                      ]
                    }
                    tintColor="#c00"
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            [
              {
                "width": "100%",
              },
              {
                "alignItems": "center",
              },
              {
                "marginBottom": 12,
                "marginTop": 12,
              },
            ]
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#eee",
                "borderColor": "#aaa",
                "borderRadius": 40,
                "borderWidth": 2,
                "height": 150,
                "paddingHorizontal": 30,
                "paddingTop": 20,
                "width": "85%",
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "column",
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "flex-start",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  style={
                    {
                      "flexDirection": "column",
                    }
                  }
                >
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#000",
                          },
                          {
                            "fontSize": 24,
                            "lineHeight": 32,
                          },
                          {
                            "fontFamily": "Quicksand-Bold",
                          },
                        ]
                      }
                    >
                      Honda
                    </Text>
                    <Image
                      resizeMode="contain"
                      source={1}
                      style={
                        [
                          {
                            "height": 40,
                          },
                          {
                            "width": 40,
                          },
                          [
                            {
                              "marginLeft": 20,
                            },
                          ],
                        ]
                      }
                      tintColor="#bbb"
                    />
                  </View>
                  <Text
                    style={
                      [
                        {
                          "fontSize": 16,
                          "lineHeight": 24,
                        },
                        {
                          "fontFamily": "Quicksand-SemiBold",
                        },
                        [
                          {
                            "marginTop": 12,
                          },
                        ],
                      ]
                    }
                  >
                    Patente: 
                    XYZ789
                  </Text>
                </View>
              </View>
              <View
                style={
                  [
                    {
                      "alignItems": "flex-end",
                    },
                    {
                      "marginBottom": 12,
                    },
                  ]
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "backgroundColor": "#eee",
                    }
                  }
                >
                  <Image
                    resizeMode="contain"
                    source={1}
                    style={
                      [
                        {
                          "height": 28,
                        },
                        {
                          "width": 28,
                        },
                      ]
                    }
                    tintColor="#c00"
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 40,
        "marginTop": 40,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 18,
            "lineHeight": 28,
          },
          {
            "color": "#4b5563",
          },
          {
            "fontFamily": "Quicksand-SemiBold",
          },
        ]
      }
    >
      Agreg√° un auto nuevo
    </Text>
    <View
      focusVisibleStyle={{}}
      focusable={true}
      href="/AddCarPage"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="link"
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomColor": "transparent",
          "borderBottomLeftRadius": 12,
          "borderBottomRightRadius": 12,
          "borderBottomWidth": 1,
          "borderLeftColor": "transparent",
          "borderLeftWidth": 1,
          "borderRightColor": "transparent",
          "borderRightWidth": 1,
          "borderStyle": "solid",
          "borderTopColor": "transparent",
          "borderTopLeftRadius": 12,
          "borderTopRightRadius": 12,
          "borderTopWidth": 1,
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "height": 32,
          "justifyContent": "center",
          "marginLeft": 12,
          "marginTop": 4,
          "paddingLeft": 18,
          "paddingRight": 18,
          "width": 32,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          [
            {
              "height": 24,
            },
            {
              "width": 24,
            },
          ]
        }
      />
    </View>
  </View>
</RNCSafeAreaView>
`;
